lib/kubernetes-deploy/kubernetes_resource/persistent_volume_claim.rb

```diff
@@ -3,16 +3,76 @@ module KubernetesDeploy
   class PersistentVolumeClaim < KubernetesResource
     TIMEOUT = 5.minutes
 
+    def sync(cache)
+      super
+      @storage_classes = cache.get_all("StorageClass").map { |sc| StorageClass.new(sc) }
+    end
+
     def status
       exists? ? @instance_data["status"]["phase"] : "Not Found"
     end
 
     def deploy_succeeded?
-      status == "Bound"
+      return false if deploy_failed?
```

Comment:
Do we need to explicitly invoke `deploy_failed?` inside `deploy_succeeded?`? I don't see this pattern anywhere else and it seems that this check will occur inside `resource_watcher.rb`, anyway ([ref](https://github.com/Shopify/kubernetes-deploy/blob/master/lib/kubernetes-deploy/resource_watcher.rb#L34)).